tests/src/Model/Log/ParsedLogIteratorTest.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2021, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Test\src\Object\Log;
+
+use Friendica\Util\ReversedFileReader;
+use Friendica\Model\Log\ParsedLogIterator;
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Parsed log iterator testing class
+ */
+class ParsedLogIteratorTest extends TestCase
+{
+	protected $pli;
+
+	public static function assertParsed($parsed, $expected_data)
+	{
+		foreach ($expected_data as $k => $v) {
+			self::assertSame($parsed->$k, $v, '"'.$k.'" does not match expectation');
+		}
+	}
+
+	protected function setUp(): void
+	{
+		$logfile = dirname(__DIR__) . DIRECTORY_SEPARATOR .
```

Comment:
The use of `DIRECTORY_SEPARATOR` is laudable but in practice verbose and useless, all relevant platforms understand `/`, Windows included.